packages/jest-matchers/src/index.js

```diff
@@ -10,27 +10,37 @@
 
 'use strict';
 
-import type {ThrowingMatcherFn, ExpectationResult} from '../types';
+import type {
+  Expect,
+  ExpectationResult,
+  ExpectationObject,
+  MatchersObject,
+  RawMatcherFn,
+  ThrowingMatcherFn,
+} from '../types';
 
 const matchers = require('./matchers');
+const GLOBAL_MATCHERS_OBJECT_KEY = '_JEST_MATCHERS';
+global[GLOBAL_MATCHERS_OBJECT_KEY] = {};
```

Comment:
I don't understand why we need this global object? Can it not just be a module-local variable?